types/streamx/index.d.ts

```diff
@@ -0,0 +1,309 @@
+// Type definitions for streamx 2.9
+// Project: https://github.com/streamxorg/streamx
+// Definitions by: Martin Heidegger <https://github.com/martinheidegger>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import { EventEmitter } from 'events';
+import { Writable as NodeJSWritable, Readable as NodeJSReadable, Stream as NodeJSStream } from 'stream';
+
+export interface AbortSignalLike {
+    addEventListener: (type: 'abort', listener: (this: AbortSignalLike, event: any) => any) => void;
+}
+
+export type AnyStream = NodeJSStream | Stream;
+export type AnyWritable<TType = any, TMap = TType, TByte = TMap> =
+    | NodeJSWritable
+    | Writable<TType, TMap, TByte>
+    | Duplex<TType, TByte, TMap, TByte>;
+export type AnyReadable<TType = any, TMap = TType, TByte = TMap> = NodeJSReadable | Readable<TType, TMap, TByte>;
+
+export interface Callback {
+    (err?: Error | null): void;
+}
+
+export interface ResultCallback<T> {
+    (err?: Error | null, result?: T): void;
+}
+
+export interface StreamOptions<TStream extends Stream<TByteType>, TByteType = any> {
+    highWaterMark?: number;
+    byteLength?: ByteLengthFunction<TStream, TByteType>;
+    open?: (this: TStream, cb: Callback) => void;
+    destroy?: (this: TStream, cb: Callback) => void;
+    predestroy?: (this: TStream, cb: Callback) => void;
+    signal?: AbortSignalLike;
+}
+
+/* tslint:disable-next-line interface-over-type-literal */
+export type StreamEvents = {
+    open: () => void;
+    close: () => void;
+    error: (error: Error) => void;
+};
+
+export type Events = { [event: string]: (...args: any[]) => void } | undefined;
+export type EventName<TEvents extends Events> = TEvents extends undefined ? string | symbol : keyof TEvents;
+export type EventListener<TEvents extends Events, TEvent extends string | symbol | number> = TEvents extends undefined
```

Comment:
Are there existing EventEmitter types that could be used here instead?

Reply:
No, I wish there were.